core/src/main/java/org/frankframework/processors/InputOutputPipeProcessor.java

```diff
@@ -150,7 +160,7 @@ protected PipeRunResult processPipe(@Nonnull PipeLine pipeLine, @Nonnull IPipe p
 					handler.setContext(pipeLineSession);
 					try {
 						XmlUtils.parseXml(inputSource, handler);
-						pipeRunResult.setResult(xmlWriter.toString());
+						pipeRunResult.setResult(xmlWriter.getWriter());
```

Comment:
Hoping this is a positive impact on performance.

Reply:
I think this is very tricky and not reliable. This relies on the embedded `Writer` in the `XmlWriter` being a `StringWriter`, on which `writer.toString()` can be called to get the proper XML message.

Of course, I do realise that `xmlWriter.getWriter()` is also just calling `writer.toString()`, which is not quite safe as that writer is sometimes a FileWriter.

I think best is to create the writer yourself then call the `.toString()` on that.

In future, we want to add some sort of MessageBuilder class that can be used to dynamically create messages from different kinds of streams.
This can then be easily plugged in place here.

Reply:
The whole reason for doing this PR, is to avoid the whole message is loaded into memory. Otherwise, we could just do the previous implementation and some String replacements (exaggeratedly expressed). So, I hoped that using the writer could work. Or this PR is an in-between state, hoping the MessageBuilder will come in the future.

Reply:
I think that at least this PR allows easily plugging in a sort of build in future.

Right now, since it uses a StringWriter in the XmlWriter, it does not yet address the memory requirements (unfortunately).

We'd need a writer that doesn't always writer everything into memory for that, but also does not always write everything to file always as that will slow down the system on small messages.